subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -619,7 +620,12 @@ public void executeTests() {
         LogLevel currentLevel = determineCurrentLogLevel();
         TestLogging levelLogging = testLogging.get(currentLevel);
         TestExceptionFormatter exceptionFormatter = getExceptionFormatter(levelLogging);
-        TestEventLogger eventLogger = new TestEventLogger(getTextOutputFactory(), currentLevel, levelLogging, exceptionFormatter);
+
+        if (buildOperationExecutor == null) {
```

Comment:
I agree but I also couldn't come up with a better solution. The setter is package-scoped so it won't be exposed to consumers. What I want to say is "replace the service with a test service" but the service registry doesn't allow you to do it.